src/clangsa/GCChecker.cpp

```diff
@@ -617,24 +637,22 @@ void GCChecker::checkBeginFunction(CheckerContext &C) const {
   // otherwise
   const auto *LCtx = C.getLocationContext();
   const auto *FD = dyn_cast<FunctionDecl>(LCtx->getDecl());
-  if (!FD)
-    return;
+  assert(FD);
+  unsigned CurrentHeight = getStackFrameHeight(C.getStackFrame());
   ProgramStateRef State = C.getState();
   bool Change = false;
   if (C.inTopFrame()) {
     State = State->set<GCDisabledAt>((unsigned)-1);
     State = State->set<SafepointDisabledAt>((unsigned)-1);
     Change = true;
   }
-  if (State->get<GCDisabledAt>() == (unsigned)-1) {
-    if (declHasAnnotation(FD, "julia_gc_disabled")) {
-      State = State->set<GCDisabledAt>(C.getStackFrame()->getIndex());
```

Comment:
@keno do you remember why you used `C.getStackFrame()->getIndex()` as the token here? That is the non-unique index into the statement in the callee's current cfg block, which was causing me problems for recursive functions. I have translated it here into the computed height of the current analysis stack instead, though we could also possibly use `getStackFrame()->getId()`, or use the `C.getStackFrame()` result directly.

Reply:
It was intended to be the depth of the stack frame, but I may have used the wrong function.

Reply:
I assumed so. However, that computation does seem to have existed before, but this should do that now.